pkg/dockerregistry/server/tagservice.go

```diff
@@ -0,0 +1,208 @@
+package server
+
+import (
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+type tagService struct {
+	distribution.TagService
+
+	repository     distribution.Repository
+	registryClient client.Interface
+	namespace      string
+	name           string
+	pullthrough    bool
```

Comment:
I think it's common for our other middlewares to reach for some item directly like `middleware.repository.<item>`.